standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/Msck.java

```diff
@@ -108,15 +108,16 @@ public int repair(MsckInfo msckInfo) {
     boolean success = false;
     long txnId = -1;
     int ret = 0;
+    long partitionExpirySeconds = msckInfo.getPartitionExpirySeconds();
     try {
       Table table = getMsc().getTable(msckInfo.getCatalogName(), msckInfo.getDbName(), msckInfo.getTableName());
       qualifiedTableName = Warehouse.getCatalogQualifiedTableName(table);
       if (getConf().getBoolean(MetastoreConf.ConfVars.MSCK_REPAIR_ENABLE_PARTITION_RETENTION.getHiveName(), false)) {
-        msckInfo.setPartitionExpirySeconds(PartitionManagementTask.getRetentionPeriodInSeconds(table));
-        LOG.info("{} - Retention period ({}s) for partition is enabled for MSCK REPAIR..",
-          qualifiedTableName, msckInfo.getPartitionExpirySeconds());
+        partitionExpirySeconds = PartitionManagementTask.getRetentionPeriodInSeconds(table);
```

Comment:
earlier code had a "sideeffect"; namely: it set mschInfo.setPartitionExpirySeconds

Reply:
Indeed, but it is not use anywhere else, but here, this is the only place where the getter is invoked. The point here was that a request type object should not be modified, it should be immutable, the request contained a certain value, which is used here like this - may be overwritten by some config value. Still the request is not modified.

Reply:
Now it is set in the constructor, the logic that overwrites the value from the config is there.